Type5 default originate

mkanjari opened the pull request:
Implemented support for default-originate command under address-family l2vpn evpn:
This is needed for the case where external routing within a POD, will follow the default route to the border/exit leaf. The border/exit leaf has more than one next hop to forward the packet on to, depending on the destination IP.

provide a match clause to match EVPN default route:
A Border Leaf can originate a default route for all the leafs within the POD.However, we do not want to advertise this route outside the POD.Therefore, we provide an option to filter a EVPN type5 default route through a route-map.

mkanjari commented:
@pguibert6WIND @riw777 @paulzlabn : Can you please help with the review when you get a chance.

pguibert6WIND commented on bgpd/bgp_evpn.h:
```diff
@@ -62,11 +62,13 @@ static inline int advertise_type5_routes(struct bgp *bgp_vrf,
-	    CHECK_FLAG(bgp_vrf->vrf_flags, BGP_VRF_ADVERTISE_IPV6_IN_EVPN))
+	    CHECK_FLAG(bgp_vrf->af_flags[AFI_L2VPN][SAFI_EVPN],
+		       BGP_L2VPN_EVPN_ADVERTISE_IPV6_UNICAST))
 		return 1;
```
is not there any conflict with DAMPENING FLAG ?

mkanjari replied:
I am not sure if we use this flag for AFI_L2VPN and SAFI_EVPN. Looks like we should change it anyways to avoid any conflicts.

mkanjari commented:
@pguibert6WIND: Thanks for catching that ! I have committed the changes which resolves the conflict in the flags.

pguibert6WIND commented:
Hi @mkanjari ,

did you remove the patch files from the commit ?

mkanjari commented:
@pguibert6WIND : I am not sure what you are referring to.

donaldsharp commented:
@mkanjari please update the PR to get the merge conflicts addressed

mkanjari commented:
@riw777 @pguibert6WIND : I have addressed the review comments and the merge conflicts. Could you please help in merging the PR  if there are no further comments ?
